packages/frontend/core/src/components/affine/auth/oauth.tsx

```diff
@@ -28,94 +33,98 @@ const OAuthProviderMap: Record<
   [OAuthProviderType.OIDC]: {
     icon: <LockIcon />,
   },
+
+  [OAuthProviderType.Apple]: {
+    icon: <AppleIcon />,
+  },
 };
 
 export function OAuth({ redirectUrl }: { redirectUrl?: string }) {
+  const t = useI18n();
   const serverService = useService(ServerService);
   const urlService = useService(UrlService);
   const oauth = useLiveData(serverService.server.features$.map(r => r?.oauth));
   const oauthProviders = useLiveData(
     serverService.server.config$.map(r => r?.oauthProviders)
   );
-  const scheme = urlService.getClientScheme();
+  const auth = useService(AuthService);
+
+  const onContinue = useAsyncCallback(
+    async (provider: OAuthProviderType) => {
+      track.$.$.auth.signIn({ method: 'oauth', provider });
+
+      const open: () => Promise<void> | void = BUILD_CONFIG.isNative
+        ? async () => {
+            try {
+              const scheme = urlService.getClientScheme();
+              const options = await auth.oauthPreflight(
+                provider,
+                scheme ?? 'web'
+              );
+              urlService.openPopupWindow(options.url);
+            } catch (e) {
+              console.error(e);
+              const err = e as UserFriendlyError;
+              notify.error({
```

Comment:
Provide a `notify.friendlyError(err)` helper method? This makes it easy to use and reduces repetitive code